core/src/Text/Interpolation/Nyan/Lens/TH.hs

```diff
@@ -0,0 +1,53 @@
+-- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io/>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+module Text.Interpolation.Nyan.Lens.TH
+  ( makeLenses
+  )
+  where
+
+import Control.Monad ((<=<))
+import Language.Haskell.TH
+
+-- | Information about the record field the lenses will operate on.
+type RecordFieldInfo = (Name, Strict, Type)
+
+-- | Given a record datatype, derives lenses for all of its fields.
+makeLenses :: Name -> Q [Dec]
+makeLenses = mapM deriveLens <=< extractRecordFields
+
+extractRecordFields :: Name -> Q [RecordFieldInfo]
+extractRecordFields datatype = do
+  let datatypeStr = nameBase datatype
+  info <- reify datatype
+  return $ case info of
+    TyConI (DataD    _ _ _ _ [RecC _ fs] _) -> fs
+    TyConI (NewtypeD _ _ _ _ (RecC _ fs) _) -> fs
+    TyConI (DataD _ _ _ _ [_] _) ->
+      error $ "Can't derive lenses without record selectors: " ++ datatypeStr
```

Comment:
Better use `fail`, `error` is for internal errors and will be reported respectively.